### PR TITLE
RoR2: some more world cleanup

### PR DESCRIPTION
**Pull request by alwaysintreble**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
just some minor cleanup of the ror2 world, such as creating items in the correct location and removing the separate rules file.

## How was this tested?
Rolled a few seeds with SoE

### CODE REVIEW

**Comment by NewSoupVi** [comment]
Lgtm.

Altho, is init eating rules an argument of "It's so short anyway"? It does feel strange to me to *reverse* a seperation of code functionality into different files like that. It feels like if anyone ever changes anything about RoR2 logic to make it more complex, they'll just be reverting this change

**Comment by alwaysintreble** [comment]
It doesn't really matter I just think `def set_rules(): set_rules()` is really bad practice. Whenever someone does change it it's probably best to have the rules file but at least use a different name for the function.

**Comment by alwaysintreble** [comment]
@kindasneaki since you're the RoR2 maintainer I guess

**Comment by kindasneaki** [comment]
I don't really understand the reasoning of removing Rules.py since it seems like there are other worlds that are doing the same thing. It would probably be easier to just rename the function if that's the main issue

File: worlds/ror2/__init__.py [hunk 1]
```diff
@@ -130,26 +159,26 @@ def create_item(self, name: str) -> Item:
...
     if total_locations / 25 == num_of_events:
         num_of_events -= 1
-    world_region = world.get_region("Petrichor V", player)
+    world_region = multiworld.get_region("Petrichor V", player)
 
     for i in range(num_of_events):
         event_loc = RiskOfRainLocation(player, f"Pickup{(i + 1) * 25}", None, world_region)
         event_loc.place_locked_item(RiskOfRainItem(f"Pickup{(i + 1) * 25}", ItemClassification.progression, None, player))
-        event_loc.access_rule(lambda state, i=i: state.can_reach(f"ItemPickup{((i + 1) * 25) - 1}", player))
+        event_loc.access_rule = lambda state, i=i: state.can_reach(f"ItemPickup{((i + 1) * 25) - 1}", player)
```
Comment (kindasneaki) [inline worlds/ror2/__init__.py:144]: This causes a error.. AssertionError: can_reach: player is required if spot is str

File: worlds/ror2/__init__.py [hunk 2]
```diff
@@ -110,14 +110,14 @@ def set_rules(self) -> None:
...
-            "itemPickupStep": self.multiworld.item_pickup_step[self.player].value,
-            "seed": "".join(self.multiworld.slot_seeds[self.player].choice(string.digits) for _ in range(16)),
-            "totalLocations": self.multiworld.total_locations[self.player].value,
-            "totalRevivals": self.multiworld.total_revivals[self.player].value,
-            "startWithDio": self.multiworld.start_with_revive[self.player].value,
-            "FinalStageDeath": self.multiworld.final_stage_death[self.player].value
-        }
+        option_results: Dict[str, Any] = {}
+        for option in ror2_options:
+            split_name = option.split(" ")
```
Comment (kindasneaki) [inline worlds/ror2/__init__.py:115]: This needs a "_" instead of " " to split the names

File: worlds/ror2/__init__.py [hunk 3]
```diff
@@ -110,14 +110,14 @@ def set_rules(self) -> None:
...
+            split_name = [name.title() for name in split_name]
+            split_name[0].lower()
```
Comment (kindasneaki) [inline worlds/ror2/__init__.py:117]: this needs to save back into split_name. 
split_name[0] = split_name[0].lower()

File: worlds/ror2/__init__.py (hunk 1 above)
Comment (alwaysintreble) [reply worlds/ror2/__init__.py:144]: I need more info on this error because I can't reproduce it. It's doing exactly as I intended.

Comment (kindasneaki) [reply worlds/ror2/__init__.py:146]: All I did was run a ror2 yaml and got the error

Comment (alwaysintreble) [reply worlds/ror2/__init__.py:145]: I wrote tests and I'm still unable to reproduce this with a 250 locations yaml
[AP_08268955892101614522_Spoiler.txt](https://github.com/ArchipelagoMW/Archipelago/files/10346215/AP_08268955892101614522_Spoiler.txt)

Comment (kindasneaki) [reply worlds/ror2/__init__.py:145]: I did a few tests in my yaml to see whats causing it and its total_locations: random-range-125-200.. Did that change somewhere recently?

File: worlds/ror2/__init__.py [hunk 4]
```diff
@@ -112,11 +112,13 @@ def set_rules(self) -> None:
     def fill_slot_data(self):
         option_results: Dict[str, Any] = {}
         for option in ror2_options:
-            split_name = option.split(" ")
-            split_name = [name.title() for name in split_name]
-            split_name[0].lower()
+            if "scrap" in option or "item" in option:
```
Comment (kindasneaki) [inline worlds/ror2/__init__.py:114]: This removes item_pickup_step which is needed in the client

File: worlds/ror2/__init__.py (hunk 1 above)
Comment (alwaysintreble) [reply worlds/ror2/__init__.py]: ok found the issue. no idea why it only came up in specific instances but it was an easy fix.

**Review by kindasneaki — APPROVED** [review]

**Pull request closed without merging**